### ISSUE TITLE
Two index constraint test sometimes fails

### ISSUE THREAD

**Issue by nilsdeppe** [sxs-collaboration/spectre#1227]
# Bug reports:

```
  CHECK( a == appx(b) )
with expansion:
  -54.0809219435 == Approx( -54.0809219434 )
with messages:
  seed: 3251596043
  function: two_index_constraint
  /work/spectre/tests/Unit/Pypp/CheckWithRandomValues.hpp:221: result == pypp::
  call<ResultType>( module_name, function_name, std::get<ArgumentIs>(args)...,
  forward_to_pypp<ResultType>(std::get<MemberArgsIs>(member_args),
  used_for_size)...)
  a: T(0,0)=(1.217964370819988893e+06,-1.353834068223283393e+05,-4.
  892301699243612529e+04,5.020769910511600901e+05)
  T(1,0)=(-1.417430402077460894e+06,3.009143172745000338e+05,-5.
  099038514102233603e+04,7.175787227195601445e+05)
  T(2,0)=(-3.180313668856734294e+05,1.216957597108158516e+06,-5.
  132668962175928755e+04,4.783539471338726580e+05)
  T(0,1)=(-1.813639959090815624e+06,-3.300650494955033064e+05,-2.
  190662904683990928e+05,1.648161863929447718e+05)
  T(1,1)=(2.165944611815787386e+06,2.208759050663033850e+05,-5.
  408092194346811254e+01,2.533025017585758469e+05)
  T(2,1)=(4.151489277041883906e+05,1.264118805408859858e+06,-2.
  476807451968216628e+05,1.539652287537635711e+05)
  T(0,2)=(-1.303967533469910035e+06,5.748343538770714076e+04,1.
  703763490008732188e+05,6.141154262538447510e+05)
  T(1,2)=(1.586021446641117102e+06,-6.672846242963812256e+04,3.
  572974716205265577e+04,8.589934286566750379e+05)
  T(2,2)=(2.713678094300511293e+05,-2.979011078643937362e+05,1.
  649637693579257175e+05,5.766036228052574443e+05)
  T(0,3)=(2.736250500648926245e+05,-3.644060540185483114e+04,5.
  264805782281181309e+05,4.456946737193198642e+05)
  T(1,3)=(-3.716414485790459439e+05,9.792238959440712642e+04,1.
  419130612863080751e+05,5.681270807716052514e+05)
  T(2,3)=(-4.855693379127969092e+04,4.127744817119947402e+05,5.
  051455970937916427e+05,3.977959522817987017e+05)
  b: T(0,0)=(1.217964370819990756e+06,-1.353834068223286886e+05,-4.
  892301699243613257e+04,5.020769910511607304e+05)
  T(1,0)=(-1.417430402077459730e+06,3.009143172744989279e+05,-5.
  099038514102233603e+04,7.175787227195608430e+05)
  T(2,0)=(-3.180313668856736040e+05,1.216957597108158050e+06,-5.
  132668962175928027e+04,4.783539471338723670e+05)
  T(0,1)=(-1.813639959090813994e+06,-3.300650494955044123e+05,-2.
  190662904683983070e+05,1.648161863929448882e+05)
  T(1,1)=(2.165944611815778539e+06,2.208759050663045491e+05,-5.
  408092194339683090e+01,2.533025017585756723e+05)
  T(2,1)=(4.151489277041876339e+05,1.264118805408859625e+06,-2.
  476807451968214300e+05,1.539652287537632219e+05)
  T(0,2)=(-1.303967533469910268e+06,5.748343538770630403e+04,1.
  703763490008731314e+05,6.141154262538442854e+05)
  T(1,2)=(1.586021446641113143e+06,-6.672846242963783152e+04,3.
  572974716205268487e+04,8.589934286566764349e+05)
  T(2,2)=(2.713678094300511875e+05,-2.979011078643944347e+05,1.
  649637693579254264e+05,5.766036228052586084e+05)
  T(0,3)=(2.736250500648924499e+05,-3.644060540185541322e+04,5.
  264805782281177817e+05,4.456946737193196896e+05)
  T(1,3)=(-3.716414485790457693e+05,9.792238959440735925e+04,1.
  419130612863077840e+05,5.681270807716054842e+05)
  T(2,3)=(-4.855693379127964727e+04,4.127744817119936342e+05,5.
  051455970937925158e+05,3.977959522817994002e+05)
  a: (2.165944611815787386e+06,2.208759050663033850e+05,-5.408092194346811254e+
  01,2.533025017585758469e+05)
  b: (2.165944611815778539e+06,2.208759050663045491e+05,-5.408092194339683090e+
  01,2.533025017585756723e+05)
  a: -5.408092194346811254e+01
  b: -5.408092194339683090e+01
2.639 s: Unit.Evolution.Systems.GeneralizedHarmonic.TwoIndexConstraint
===============================================================================
test cases:    1 |    0 passed | 1 failed
assertions: 6604 | 6603 passed | 1 failed
```

This happened on Travis for #1139

**Comment by nilsdeppe** [comment]
And another one...
```
 CHECK( a == appx(b) )
with expansion:
  51.8093949662 == Approx( 51.8093949661 )
with messages:
  seed: 1042150812
  function: two_index_constraint
  /work/spectre/tests/Unit/Pypp/CheckWithRandomValues.hpp:221: result == pypp::
  call<ResultType>( module_name, function_name, std::get<ArgumentIs>(args)...,
  forward_to_pypp<ResultType>(std::get<MemberArgsIs>(member_args),
  used_for_size)...)
  a: T(0,0)=(-3.860495940049516776e+03,-6.063033555845018709e+05,-5.
  714731504448906053e+05,-1.634598249848070322e+06)
  T(1,0)=(-1.059396532735157234e+04,4.934676161605585949e+05,6.
  018951134404935874e+05,2.434552419094817888e+05)
  T(2,0)=(5.759547956355256247e+03,1.062155083506210009e+06,-1.
  834525124619925045e+05,-2.627417100338205229e+06)
  T(0,1)=(2.016000441512829275e+04,4.126869907575332327e+05,1.
  873786683819330938e+05,3.401365536968114320e+06)
  T(1,1)=(2.167086878628376144e+04,-3.386191991657534381e+05,-1.
  475684592592615809e+05,-4.991412656578224851e+05)
  T(2,1)=(-3.097717145898151557e+03,-7.194717322124341736e+05,9.
  499002643348999845e+04,5.439623897334646434e+06)
  T(0,2)=(2.390388395111423597e+04,3.032724905972792185e+05,1.
  010739969111473765e+06,-1.857713089866091497e+06)
  T(1,2)=(5.066423579503931978e+04,-2.237398542689704918e+05,-1.
  024904291424092953e+06,3.069221176105277264e+05)
  T(2,2)=(2.265971846615170216e+04,-4.811547130329312058e+05,3.
  280732865443933406e+05,-3.083171907689663116e+06)
  T(0,3)=(-3.143382194655428611e+04,-1.097241981402211823e+05,8.
  383092937704577344e+05,-4.603493926611323841e+06)
  T(1,3)=(-2.625985105086744443e+04,7.507561849019510555e+04,-8.
  309041476754149189e+05,7.130432636709456565e+05)
  T(2,3)=(5.180939496621395790e+01,1.832603091652564763e+05,2.
  886456099596275017e+05,-7.469173326837778091e+06)
  b: T(0,0)=(-3.860495940049507681e+03,-6.063033555845002411e+05,-5.
  714731504448928172e+05,-1.634598249848068459e+06)
  T(1,0)=(-1.059396532735160326e+04,4.934676161605587113e+05,6.
  018951134404917248e+05,2.434552419094811776e+05)
  T(2,0)=(5.759547956355264432e+03,1.062155083506212104e+06,-1.
  834525124619923881e+05,-2.627417100338210352e+06)
  T(0,1)=(2.016000441512834004e+04,4.126869907575332909e+05,1.
  873786683819330065e+05,3.401365536968111992e+06)
  T(1,1)=(2.167086878628347404e+04,-3.386191991657520412e+05,-1.
  475684592592619883e+05,-4.991412656578244641e+05)
  T(2,1)=(-3.097717145898245690e+03,-7.194717322124348721e+05,9.
  499002643348983838e+04,5.439623897334638052e+06)
  T(0,2)=(2.390388395111448699e+04,3.032724905972793349e+05,1.
  010739969111474231e+06,-1.857713089866097784e+06)
  T(1,2)=(5.066423579503921792e+04,-2.237398542689704045e+05,-1.
  024904291424088296e+06,3.069221176105276681e+05)
  T(2,2)=(2.265971846615175673e+04,-4.811547130329300999e+05,3.
  280732865443944465e+05,-3.083171907689656597e+06)
  T(0,3)=(-3.143382194655413332e+04,-1.097241981402212259e+05,8.
  383092937704571523e+05,-4.603493926611313596e+06)
  T(1,3)=(-2.625985105086748445e+04,7.507561849019535293e+04,-8.
  309041476754115429e+05,7.130432636709461221e+05)
  T(2,3)=(5.180939496608166195e+01,1.832603091652569128e+05,2.
  886456099596269196e+05,-7.469173326837769710e+06)
  a: (5.180939496621395790e+01,1.832603091652564763e+05,2.886456099596275017e+
  05,-7.469173326837778091e+06)
  b: (5.180939496608166195e+01,1.832603091652569128e+05,2.886456099596269196e+
  05,-7.469173326837769710e+06)
  a: 5.180939496621395790e+01
  b: 5.180939496608166195e+01
0.181 s: Unit.Evolution.Systems.GeneralizedHarmonic.TwoIndexConstraint
===============================================================================
test cases:    1 |    0 passed | 1 failed
assertions: 6604 | 6603 passed | 1 failed
```

This is GCC-6 release on Travis

**Comment by nilsdeppe** [comment]
Marking as a priority because as long as this is broken it seems we'll have a lot of travis builds fail.

**Comment by nilsdeppe** [comment]
Here's another case:

```
 CHECK( a == appx(b) )
with expansion:
  -20.0917226985 == Approx( -20.0917226984 )
with messages:
  seed: 1663355070
  function: two_index_constraint
  /work/spectre/tests/Unit/Pypp/CheckWithRandomValues.hpp:221: result == pypp::
  call<ResultType>( module_name, function_name, std::get<ArgumentIs>(args)...,
  forward_to_pypp<ResultType>(std::get<MemberArgsIs>(member_args),
  used_for_size)...)
  a: T(0,0)=(9.738285561159995268e+04,5.607697457355071092e+05,-2.
  459992221692405001e+05,2.942546528220472392e+06)
  T(1,0)=(-6.002320985502277035e+05,-6.090366752579397289e+05,-8.
  389660244760395726e+05,7.222255923543506069e+05)
  T(2,0)=(-7.313092641843732963e+03,2.555447645275184914e+04,4.
  341015481576974853e+05,6.426483207391330507e+04)
  T(0,1)=(-8.795080231547633593e+04,-4.285622809429349727e+05,-2.
  195101088870044332e+05,2.119666759409020189e+06)
  T(1,1)=(6.800580903782708338e+05,3.897904083303173538e+05,-8.
  319304103947067633e+05,5.318049365886783926e+05)
  T(2,1)=(-5.471346279948256779e+03,-2.009172269848204451e+01,4.
  449040366946322611e+05,9.308397470389748923e+04)
  T(0,2)=(-3.624061295446350414e+04,1.148628718828819925e+06,3.
  316764177090370213e+05,-1.464923430087479064e+06)
  T(1,2)=(4.063730204824533430e+05,-1.227989178434347967e+06,1.
  105764356750899227e+06,-3.371688926139331306e+05)
  T(2,2)=(-1.934507772652036147e+04,5.696842881257608678e+04,-6.
  136351760008876445e+05,1.790060420327848078e+03)
  T(0,3)=(1.446252832664054004e+05,8.634728791052809684e+05,-3.
  522673620796935866e+05,-1.527480897859492106e+06)
  T(1,3)=(-8.948710471159134759e+05,-9.482302500984123908e+05,-1.
  159825867833199911e+06,-3.956279084069519304e+05)
  T(2,3)=(-1.171184047261279557e+04,4.590528794459522032e+04,6.
  682254238117487403e+05,-3.739452834014227119e+04)
  b: T(0,0)=(9.738285561159935605e+04,5.607697457355073420e+05,-2.
  459992221692403255e+05,2.942546528220480774e+06)
  T(1,0)=(-6.002320985502292169e+05,-6.090366752579410095e+05,-8.
  389660244760399219e+05,7.222255923543478129e+05)
  T(2,0)=(-7.313092641843671117e+03,2.555447645275162358e+04,4.
  341015481576957973e+05,6.426483207391419273e+04)
  T(0,1)=(-8.795080231547684525e+04,-4.285622809429347981e+05,-2.
  195101088870046078e+05,2.119666759409019724e+06)
  T(1,1)=(6.800580903782700188e+05,3.897904083303165971e+05,-8.
  319304103947047843e+05,5.318049365886794403e+05)
  T(2,1)=(-5.471346279948299525e+03,-2.009172269838541069e+01,4.
  449040366946316790e+05,9.308397470389588852e+04)
  T(0,2)=(-3.624061295446362055e+04,1.148628718828817364e+06,3.
  316764177090364974e+05,-1.464923430087479763e+06)
  T(1,2)=(4.063730204824535176e+05,-1.227989178434352390e+06,1.
  105764356750895502e+06,-3.371688926139330724e+05)
  T(2,2)=(-1.934507772652029234e+04,5.696842881257668341e+04,-6.
  136351760008879937e+05,1.790060420327820339e+03)
  T(0,3)=(1.446252832664047601e+05,8.634728791052824818e+05,-3.
  522673620796933537e+05,-1.527480897859493038e+06)
  T(1,3)=(-8.948710471159089357e+05,-9.482302500984084327e+05,-1.
  159825867833198281e+06,-3.956279084069514065e+05)
  T(2,3)=(-1.171184047261263913e+04,4.590528794459508208e+04,6.
  682254238117473433e+05,-3.739452834013969550e+04)
  a: (-5.471346279948256779e+03,-2.009172269848204451e+01,4.
  449040366946322611e+05,9.308397470389748923e+04)
  b: (-5.471346279948299525e+03,-2.009172269838541069e+01,4.
  449040366946316790e+05,9.308397470389588852e+04)
  a: -2.009172269848204451e+01
  b: -2.009172269838541069e+01
1.708 s: Unit.Evolution.Systems.GeneralizedHarmonic.TwoIndexConstraint
===============================================================================
test cases:    1 |    0 passed | 1 failed
assertions: 6604 | 6603 passed | 1 failed
```

**Comment by geoffrey4444** [comment]
Sorry for somehow missing this earlier! I'll regularly check the issues from now on. I'll deal with this ASAP.

**Comment by kidder** [comment]
Fixed by #1245
